### PR TITLE
[Numpy] ffi invocation: expand_dims, tril, diff, broadcast_to

### PR DESCRIPTION
**Pull request by Alicia1529**
## Description ##
as title

### CODE REVIEW

File: src/api/operator/numpy/matrix_op.cc [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file matrix_op.cc
```
Comment (haojin2) [inline src/api/operator/numpy/matrix_op.cc]: better rename the file to `np_matrix_op.cc` to keep the consistency.

File: src/api/operator/numpy/np_diff_op.cc [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+namespace mxnet {
+
+MXNET_REGISTER_API("_npi.diff")
+.set_body([](runtime::MXNetArgs args, runtime::MXNetRetValue* ret) {
+  using namespace runtime;
+  const nnvm::Op* op = Op::Get("_npi_diff");
+  nnvm::NodeAttrs attrs;
+  op::DiffParam param;
+  param.n = args[1].operator int64_t();
```
Comment (hzfan) [inline src/api/operator/numpy/np_diff_op.cc]: Why not `operator int()` ?

File: src/api/operator/numpy/matrix_op.cc [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+namespace mxnet {
+
+MXNET_REGISTER_API("_npi.expand_dims")
+.set_body([](runtime::MXNetArgs args, runtime::MXNetRetValue* ret) {
+  using namespace runtime;
+  const nnvm::Op* op = Op::Get("_npi_expand_dims");
+  nnvm::NodeAttrs attrs;
+  op::ExpandDimParam param;
+  param.axis = args[1].operator int64_t();
```
Comment (hzfan) [inline src/api/operator/numpy/matrix_op.cc]: `param.axis` is int, seems that `operator int()` is more reasonable here.

File: src/api/operator/numpy/matrix_op.cc [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+  attrs.parsed = std::move(param);
```
Comment (hzfan) [inline src/api/operator/numpy/matrix_op.cc]: `ExpandDim` contains just one int. No need to std::move here.
Suggested change:


```
// we directly copy ExpandDimParam, which is trivially-copyable
attrs.parsed = param;
```

File: src/api/operator/numpy/np_diff_op.cc [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  param.axis = args[2].operator int64_t();
+
+  attrs.parsed = std::move(param);
```
Comment (hzfan) [inline src/api/operator/numpy/np_diff_op.cc]: Same as above. Please remove std::move and add a comment.

File: src/api/operator/numpy/np_tril_op.cc [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+MXNET_REGISTER_API("_npi.tril")
+.set_body([](runtime::MXNetArgs args, runtime::MXNetRetValue* ret) {
+  using namespace runtime;
+  const nnvm::Op* op = Op::Get("_npi_tril");
+  nnvm::NodeAttrs attrs;
+  op::TrilParam param;
+  param.k = args[1].operator int64_t();
+
+  attrs.parsed = std::move(param);
```
Comment (hzfan) [inline src/api/operator/numpy/np_tril_op.cc]: Same as above. Please remove std::move and add a comment.

File: src/api/operator/numpy/np_broadcast_reduce_op_value.cc [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+MXNET_REGISTER_API("_npi.broadcast_to")
+.set_body([](runtime::MXNetArgs args, runtime::MXNetRetValue* ret) {
+  using namespace runtime;
+  const nnvm::Op* op = Op::Get("_npi_broadcast_to");
+  nnvm::NodeAttrs attrs;
+  op::BroadcastToParam param;
+  if (args[1].type_code() == kDLInt) {
+      param.shape = TShape(1, args[1].operator int64_t());
+    } else {
```
Comment (hzfan) [inline src/api/operator/numpy/np_broadcast_reduce_op_value.cc]: Please align.

File: src/api/operator/numpy/np_broadcast_reduce_op_value.cc [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+  } else {
+    param.shape = TShape(args[1].operator ObjectRef());
+  }
+  attrs.parsed = std::move(param);
+  attrs.op = op;
+  SetAttrDict<op::BroadcastToParam>(&attrs);
+
+  int num_outputs = 0;
+  NDArray* inputs[] = {args[0].operator mxnet::NDArray*()};
+  auto ndoutputs = Invoke(op, &attrs, 1, inputs, &num_outputs, nullptr);
```
Comment (hzfan) [inline src/api/operator/numpy/np_broadcast_reduce_op_value.cc]: The `1` seems like a magic number. Better to use `int num_inputs = 1` and call `Invoke` with `num_inputs`.  Same for 3 other operators.

**Pull request merged**